docs/root/start/sandboxes/cors.rst

```diff
@@ -0,0 +1,117 @@
+    data from this endpoint.
+  * Restricted: CORS is enabled on the route requested and the only allowed
+    origin is ``envoyproxy.io``. This will result in a client-side CORS error.
+
+Running the Sandboxes
+~~~~~~~~~~~~~~~~~~~~~
+
+The following documentation runs through the setup of both services.
+
+**Step 1: Install Docker**
+
+Ensure that you have a recent versions of ``docker``, ``docker-compose`` and
+``docker-machine`` installed.
+
+A simple way to achieve this is via the `Docker Toolbox <https://www.docker.com/products/docker-toolbox>`_.
+
+**Step 2: Setup Docker Machines**
+
+First, let's create a couple of new machines which will hold the containers.
+
+Terminal 1
+
+.. code-block:: console
+
+  $ docker-machine create --driver virtualbox frontend
+  $ eval $(docker-machine env frontend)
+
+Terminal 2
+
+.. code-block:: console
+
+  $ docker-machine create --driver virtualbox backend
+  $ eval $(docker-machine env backend)
+
+**Step 3: Clone the Envoy repo and start all of our containers**
+
+If you have not cloned the Envoy repo, clone it with ``git clone git@github.com:envoyproxy/envoy``
+or ``git clone https://github.com/envoyproxy/envoy.git``
+
+Terminal 1
+
+.. code-block:: console
+
+  $ pwd
+  envoy/examples/cors/frontend
+  $ docker-compose up --build -d
+  $ docker-compose ps
+          Name                                  Command                    State                  Ports
+  -----------------------------------------------------------------------------------------------------------------------------------------
+  frontend_front-envoy_1_cb51c62edc96        /usr/bin/dumb-init -- /bin ... Up      10000/tcp, 0.0.0.0:8000->80/tcp, 0.0.0.0:8001->8001/tcp
+  frontend_frontend-service_1_491cf87432cd   /bin/sh -c /usr/local/bin/ ... Up      10000/tcp, 80/tcp
+
+Terminal 2
+
+.. code-block:: console
+
+  $ pwd
+  envoy/examples/cors/backend
+  $ docker-compose up --build -d
+  $ docker-compose ps
+          Name                                  Command                    State                  Ports
+  -----------------------------------------------------------------------------------------------------------------------------------------
+  backend_front-envoy_1_7f9d5039c86f       /usr/bin/dumb-init -- /bin ... Up      10000/tcp, 0.0.0.0:8000->80/tcp, 0.0.0.0:8001->8001/tcp
+  backend_backend-service_1_c7752ae7192a   /bin/sh -c /usr/local/bin/ ... Up      10000/tcp, 80/tcp
+
+**Step 4: Test Envoy's CORS capabilities**
+
+You can now open a browser to view your frontend service. To find the IP of
+your frontend service run the following command in terminal 1.
+
+.. code-block:: console
+
+  $ docker-machine ip frontend
+
+In terminal 2, run the same command to get the remote origin IP that the
+frontend service will be making requests to.
+
+Results of the cross-origin request will be shown on the page under *Request Results*.
+Your browser's CORS enforcement logs can be found in the console.
```

Comment:
One last thing. It'd be good to add some results and sample logs for the 3 cases here.

Reply:
I've included the console logs for CORS errors. For the two failure cases the errors are the same and for the success case, there are no console errors.